specification/marketplacecatalog/data-plane/Reviews/preview/2023-01-01-preview/reviews.json

```diff
@@ -0,0 +1,405 @@
+        },
+        "isvReply": {
+          "$ref": "#/definitions/IsvReplyDto"
+        },
+        "createdDateTime": {
+          "type": "string",
+          "format": "date-time"
+        },
+        "lastModifiedDateTime": {
+          "type": "string",
+          "format": "date-time"
+        },
+        "commentsCount": {
+          "type": "integer",
+          "format": "int32"
+        },
+        "helpfulCount": {
+          "type": "integer",
+          "format": "int32"
+        }
+      }
+    },
+    "IsvReplyDto": {
+      "type": "object",
+      "required": [
+        "repliedDateTime"
+      ],
+      "properties": {
+        "content": {
+          "type": "string"
+        },
+        "repliedDateTime": {
+          "type": "string",
+          "format": "date-time"
+        }
+      }
+    },
+    "UserMetadataDto": {
+      "type": "object",
+      "required": [
+        "isEligible"
+      ],
+      "properties": {
+        "review": {
+          "$ref": "#/definitions/ReviewDto"
+        },
+        "helpfulReviewIds": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "commentReviewIds": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "isEligible": {
+          "$ref": "#/definitions/EligibilityStatus"
+        }
+      }
+    },
+    "EligibilityStatus": {
+      "type": "string",
+      "description": "",
+      "x-enumNames": [
+        "Eligible",
+        "NotEligible",
+        "PendingCheck"
+      ],
+      "enum": [
+        "Eligible",
+        "NotEligible",
+        "PendingCheck"
+      ]
+    },
+    "UpsertReviewRequest": {
+      "type": "object",
+      "description": "Create or update review request object for a product",
+      "required": [
+        "rating"
+      ],
+      "properties": {
+        "uniqueProductId": {
+          "type": "string",
+          "description": "Product unique identifier"
+        },
+        "rating": {
+          "type": "integer",
+          "description": "Rating for this review. Possible values between 1 to 5",
+          "format": "int32"
+        },
+        "title": {
+          "type": "string",
+          "description": "Title for this review"
+        },
+        "content": {
+          "type": "string",
+          "description": "Content for this review"
+        },
+        "isAnonymous": {
+          "type": "boolean",
+          "description": "Publish review as anonymous"
+        },
+        "sendEmailConsent": {
+          "type": "boolean",
+          "description": "Consent to use email"
+        }
+      }
+    },
+    "PageResultOfReviewCommentDto": {
+      "type": "object",
+      "required": [
+        "count"
+      ],
+      "properties": {
+        "items": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ReviewCommentDto"
+          }
+        },
+        "count": {
```

Comment:
We don't usually include `count` in a pageable response, what if an item gets added while a client is iterating over the paged response?

Reply:
known issue and client should expect and handle it correctly.

Reply:
I've phrased the issue with including `count` poorly. If you include `count` now and in the future your list of items grow very large you lose the flexibility of paging because your server needs to compute the total count even for only the first page. That's why we usually recommend strongly to not include a `count` property. Your case might be different if your service keeps an up-to-date precomputed totalCount value somewhere but please consider to just remove it now and only add it if customers ask for it.